app/models/like.rb

```diff
@@ -1,2 +1,9 @@
 class Like < ApplicationRecord
+  # Relationship
+  belongs_to :user, class_name: 'User', foreign_key: 'user_id'
+  belongs_to :post, class_name: 'Post', foreign_key: 'post_id'
+
+  def update_like_count
+    post.update(likes_counter: post.likes.count)
+  end
```

Comment:
- [ ] The same thing for the like model 👏 . I will also suggest you to use `after_save` to call the `#update_like_count` method every time a new like has been saved to database, then make the `update_like_count` method private, It will optimize the work 👍 

 ```ruby
...
  after_save :update_like_count

  private

  def update_like_count
   post.update(likes_counter: post.likes.count)
  end
...
```